pkg/controller/vsphere/reconciler.go

```diff
@@ -698,7 +705,7 @@ func taskIsFinished(task *mo.Task) (bool, error) {
 	case types.TaskInfoStateSuccess:
 		return true, nil
 	case types.TaskInfoStateError:
-		return true, nil
```

Comment:
why this change?

Reply:
We were not reporting an error before that, so even getting this `invalid operation for device '0'` was possible only from vmware ui. I did this as otherwise we lose similar errors.

Reply:
Oh, and the following `update` call on the machine fails with the inability to find VM by UUID, because the error here causes the VM to be removed automatically. This results in a nil panic in the code, and the controller goes into a crash loop.

Reply:
thanks, please move to a different commit.

Reply:
i'm having trouble following the conversation here, should this be moved to another PR or not?

Reply:
@elmiko The change is already in its own commit, so this PR is just a holder for both atm. Also Joel hit this issue, and we discussed that the fix will be contained in this PR https://coreos.slack.com/archives/GE2HQ9QP4/p1599829591211300?thread_ts=1599829162.210500&channel=GE2HQ9QP4&message_ts=1599829591.211300